components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/dao/SAMLSSOServiceProviderDAO.java

```diff
@@ -39,33 +122,210 @@ public SAMLSSOServiceProviderDAO(Registry registry) {
         this.registry = registry;
     }
 
-    protected SAMLSSOServiceProviderDO resourceToObject(Resource resource) {
-        SAMLSSOServiceProviderDO serviceProviderDO = new SAMLSSOServiceProviderDO();
-        serviceProviderDO.setIssuer(resource
-                .getProperty(IdentityRegistryResources.PROP_SAML_SSO_ISSUER));
-        serviceProviderDO.setAssertionConsumerUrl(resource
-                .getProperty(IdentityRegistryResources.PROP_SAML_SSO_ASSERTION_CONS_URL));
-        serviceProviderDO.setCertAlias(resource
-                .getProperty(IdentityRegistryResources.PROP_SAML_SSO_ISSUER_CERT_ALIAS));
+    /**
+     * generate an opensaml metadata object from a metadata string
+     *
+     * @param metadataString
+     * @return
+     */
+    private EntityDescriptor generateMetadataObjectFromString(String metadataString) {
+        EntityDescriptor entityDescriptor = null;
 
-        if (resource
-                .getProperty(IdentityRegistryResources.PROP_SAML_SSO_USE_FULLY_QUALIFIED_USERNAME_AS_SUBJECT) != null) {
-            serviceProviderDO
-                    .setUseFullyQualifiedUsername(new Boolean(
-                            resource.getProperty(
-                                    IdentityRegistryResources.PROP_SAML_SSO_USE_FULLY_QUALIFIED_USERNAME_AS_SUBJECT)
-                                    .trim()));
+        try {
+            Configuration.registerObjectProvider(UseFullyQualifiedUsername.DEFAULT_ELEMENT_NAME, new
```

Comment:
Normally, as a best practice and for readability, we use line break symbols or characters such as comma or full stop punctuation (',' , '.') to break a line go to the next within our java source. Please fix these accordingly throughout all your PRs.

For example, the above line , you can break the line into multiple like the following. Configuration.registerObjectProvider(UseFullyQualifiedUsername.DEFAULT_ELEMENT_NAME,
                new UseFullyQualifiedUsernameBuilder(), new UseFullyQualifiedUsernameMarshaller(),
                new UseFullyQualifiedUsernameUnmarshaller());